verifiable-api/server/bin/handlers.rs

```diff
@@ -0,0 +1,439 @@
+        .await
+        .map_err(map_server_err)?;
+
+    let code = rpc
+        .get_code(address, block_num)
+        .await
+        .map_err(map_server_err)?;
+
+    Ok(Json(AccountResponse {
+        account: Account {
+            balance: proof.balance,
+            nonce: proof.nonce,
+            code_hash: proof.code_hash,
+            storage_root: proof.storage_hash,
+        },
+        code: code.into(),
+        account_proof: proof.account_proof,
+        storage_proof: proof.storage_proof,
+    }))
+}
+
+/// This method returns the receipt of a transaction along with a Merkle proof of its inclusion.
+///
+/// Replaces the `eth_getTransactionReceipt` RPC method.
+pub async fn get_transaction_receipt<N: NetworkSpec, R: ExecutionRpc<N>>(
+    Path(tx_hash): Path<B256>,
+    State(ApiState { rpc, .. }): State<ApiState<N, R>>,
+) -> Response<Option<TransactionReceiptResponse<N>>> {
+    let receipt = rpc
+        .get_transaction_receipt(tx_hash)
+        .await
+        .map_err(map_server_err)?;
+    if receipt.is_none() {
+        // ToDo(@eshaan7): Maybe throw a 404 instead?
+        return Ok(Json(None));
+    }
+    let receipt = receipt.unwrap();
+
+    let receipts = rpc
+        .get_block_receipts(BlockId::from(receipt.block_number().unwrap()))
+        .await
+        .map_err(map_server_err)?
+        .ok_or_else(|| {
+            (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                json_err("No receipts found for the block"),
+            )
+        })?;
+
+    let receipt_proof =
+        create_receipt_proof::<N>(receipts, receipt.transaction_index().unwrap() as usize);
+
+    Ok(Json(Some(TransactionReceiptResponse {
+        receipt,
+        receipt_proof,
+    })))
+}
+
+/// This method returns an array of all logs matching the given filter.
+/// Corresponding to each log, it also returns the transaction receipt and a Merkle proof of its inclusion.
+///
+/// Replaces the `eth_getLogs` RPC method.
+pub async fn get_logs<N: NetworkSpec, R: ExecutionRpc<N>>(
+    Query(logs_filter_query): Query<LogsQuery>,
+    State(ApiState { rpc, .. }): State<ApiState<N, R>>,
+) -> Response<LogsResponse<N>> {
+    let filter: Filter = logs_filter_query.try_into().map_err(map_server_err)?;
+
+    // Fetch the filter logs from RPC
+    let logs = rpc.get_logs(&filter).await.map_err(map_server_err)?;
+
+    // Create receipt proofs for each log
+    let receipt_proofs = create_receipt_proofs_for_logs(&logs, rpc)
+        .await
+        .map_err(map_server_err)?;
+
+    Ok(Json(LogsResponse {
+        logs,
+        receipt_proofs,
+    }))
+}
+
+/// This method returns an array of all logs matching the filter with given id.
+/// Corresponding to each log, it also returns the transaction receipt and a Merkle proof of its inclusion.
+///
+/// Replaces the `eth_getFilterLogs` RPC method.
+pub async fn get_filter_logs<N: NetworkSpec, R: ExecutionRpc<N>>(
+    Path(filter_id): Path<U256>,
```

Comment:
What if our rpc doesn't have the filter id? Maybe we could try handling filter ID in Helios and convert it into some representation that doesn't have it? Just thinking about some world where this thing is registering tons of filters to its execution client, but then the client gets reset or something.

There's probably a few ways we could try and handle it.

Reply:
I'm mostly wondering if there is some improvement we can make to try and avoid the case where the rpc doesn't have the filter installed. Not sure if there is anything better we could do but lets think on it.

Reply:
One possible improvement can be to remove the filter from Helios' state if RPC returns "filter not found" error. So subsequent requests for the same filter ID won't even hit the RPC.